src/main.js

```diff
@@ -114,56 +115,91 @@ class Game {
   }
 
   imageOnClick(idClicked) {
-    if (idClicked === this.prevClicked) return;
-    if (this.openedCards === 2) return;
-    this.cards[idClicked].open();
-    if (this.openedCards === 1) {
-      this.prevClicked = idClicked;
-      return;
+    if (this.fieldIsBlocked) return;
+    switch (this.openedCardsNum) {
+      case 0:
+        this.cards[idClicked].open();
+        return;
+      case 1:
+        if (this.isCardOpened(idClicked)) return;
+        this.cards[idClicked].open();
+        this.fieldIsBlocked = true;
+        this.checkCards();
+        break;
+      case 2:
+        break;
+      default:
     }
-    if (!this.cardsEqual(this.prevClicked, idClicked)) {
-      setTimeout(() => this.returnBackImage(idClicked), 1000);
+  }
+
+  isCardOpened(cardId) {
+    return parseInt(this.openedCards[0].div.id.slice(3), 10) === cardId;
+  }
+
+  checkCards() {
+    if (!this.areCardsEqual()) {
+      setTimeout(() => this.returnBackImage(), 1000);
       return;
     }
-    setTimeout(() => {
-      this.cards[idClicked].hide();
-      this.cards[this.prevClicked].hide();
-      this.prevClicked = undefined;
-      if (!this.visibleCards) {
-        this.timer.stopTimer();
-      }
-    }, 600);
+    setTimeout(() => this.hideGuessedCards(), 600);
   }
 
-  cardsEqual(cardIndex1, cardIndex2) {
-    return this.cards[cardIndex1].hiddenValue === this.cards[cardIndex2].hiddenValue;
+  areCardsEqual() {
+    const hiddenValuesOfOpened = this.cards.filter(card => card.opened)
+      .map(card => card.hiddenValue);
+    return hiddenValuesOfOpened[0] === hiddenValuesOfOpened[1];
   }
 
-  returnBackImage(idClicked) {
-    this.cards[idClicked].close();
-    this.cards[this.prevClicked].close();
-    this.prevClicked = undefined;
+  returnBackImage() {
+    const { openedCards } = this;
+    openedCards[0].close();
+    openedCards[1].close();
+    this.fieldIsBlocked = false;
   }
 
-  get openedCards() {
+  hideGuessedCards() {
+    const { openedCards } = this;
+    openedCards[0].hide();
+    openedCards[1].hide();
+    this.fieldIsBlocked = false;
+    if (!this.visibleCardsNum) {
+      this.timer.stopTimer();
+    }
+  }
+
+  get openedCardsNum() {
     return this.cards.filter(a => a.opened).length;
   }
 
-  get visibleCards() {
+  get openedCards() {
+    return this.cards.filter(a => a.opened);
+  }
+
+  get visibleCardsNum() {
     return this.cards.filter(a => a.visible).length;
   }
 }
 
 
 class DOMManager {
   constructor() {
-    this.clickListenerBinded = this.clickListener.bind(this);
-    this.submitOnClickBinded = this.submitOnClick.bind(this);
+    this.cardOnClick = (event) => {
```

Comment:
It's not clear for me why you decided to move it here. Just to avoid `bind` ?
As for me this particular change is worsened the readability

Reply:
I replaced a function and a variable with arrow function. Look at this:
previosly:
`this.clickListenerBinded` variable,
`this.clickListener` function
and now:
`this.cardOnClick` variable

I think, in this way code become simpler.

Reply:
What about this? 🙂 
```js
this.clickListener = this.clickListener.bind(this);
```